app/src/main/java/com/example/dine_and_donate/HomeFragments/NotificationsAdapter.java

```diff
@@ -0,0 +1,116 @@
+        mDatabase = FirebaseDatabase.getInstance();
+        mRef = mDatabase.getReference();
+
+        View feedView = inflater.inflate(R.layout.item_notification, parent, false);
+        ViewHolder holder = new ViewHolder((feedView));
+        return holder;
+    }
+
+    //bind the values based on the position of the element, called as a user scrolls down
+    public void onBindViewHolder(final ViewHolder holder, int position) {
+        //get data according to position
+        Notifications notification = mNotifications.get(position);
+        String eventId = notification.getEventId();
+        String yelpId = notification.getYelpId();
+        mNotificationsRef = mRef.child("events").child(yelpId).child(eventId);
+        mNotificationsRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                holder.mEventName.setText(dataSnapshot.child("info").getValue().toString());
+                holder.mStartDate.setText(dataSnapshot.child("startTime").getValue().toString());
+                holder.mPartner.setText(dataSnapshot.child("locationString").getValue().toString());
+            }
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
+
+            }
+        });
+    }
+
+    @Override
+    //must return the size; if it is 0, nothing will render on the screen
+    public int getItemCount() {
+        Log.d("adapter", "size " + mNotifications.size());
+        return mNotifications.size();
+    }
+
+    // Clean all elements of the recycler
+    public void clear() {
+        mNotifications.clear();
+        notifyDataSetChanged();
+    }
+
+    // Add a list of items -- change to type used
+    public void addAll(List<Notifications> list) {
+        mNotifications.addAll(list);
+        notifyDataSetChanged();
+    }
+
+    //create ViewHolder pattern that will contain all the fine view by ID lookups
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        public ImageView mOrgPic;
+        public TextView mEventName;
+        public TextView mStartDate;
+        public TextView mPartner;
+        public TextView mOrgName;
+
+        //constructor takes in an inflated layout
+        public ViewHolder(View itemView) {
+            //call the superclass
+            super(itemView);
+            //perform findViewById lookups
+            mOrgPic = (ImageView) itemView.findViewById(R.id.org_iv);
+            mEventName = (TextView) itemView.findViewById(R.id.event_name_tv);
+            mStartDate = (TextView) itemView.findViewById(R.id.date_tv);
+            mPartner = (TextView) itemView.findViewById(R.id.partnered_with_tv);
+            mOrgName = (TextView) itemView.findViewById(R.id.org_name_tv);
+        }
+
+        @Override
+        //in case we want to make the notification clickable
```

Comment:
you probably do right? i would put 

// TODO: make notification clickable

and make sure to add it to your plans for work :)